packages/codemods/src/v8-v9/transforms/subComponents.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { API, FileInfo } from 'jscodeshift';
+import {
+  JsCodeshiftDefaultOptions,
+  SubComponentPropertiesOptions,
+  SubComponentsTransformIds
+} from '../../types';
+import { getTransformOptions } from './utils';
+import { updateSubComponentPropertiesTransform } from './utils/updateSubComponentProperties';
+
+const OPTIONS: Record<SubComponentsTransformIds, SubComponentPropertiesOptions> = {
+  'forms-subComponents': {
+    importSource: '@zendeskgarden/react-forms',
+    subComponents: ['Hint', 'Label', 'Message'],
+    mainComponent: 'Field'
+  },
+  'grid-subComponents': {
+    importSource: '@zendeskgarden/react-grid',
+    subComponents: ['Col', 'Row'],
+    mainComponent: 'Grid'
+  },
+  'modals-subComponents': {
+    importSource: '@zendeskgarden/react-modals',
+    subComponents: ['Body', 'Close', 'Footer', 'FooterItem', 'Header'],
+    mainComponent: 'Modal'
+  },
+  'alert-subComponents': {
+    importSource: '@zendeskgarden/react-notifications',
+    subComponents: ['Close', 'Paragraph', 'Title'],
+    mainComponent: 'Well',
+    bannedComponents: ['Well', 'Notification']
```

Comment:
`Alert`, `Well`, and `Notifications` share the same subcomponent names. To avoid modifying JSX nodes incorrectly, it's preferable to skip files that import a combination of these components.